### PR TITLE
fix(manager/helmv3): ignore implied local repositories 

### LINKED ISSUE #14261: helmv3: Renovate cannot handle local dependencies anymore
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

31.80.1

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab Community Edition 13.12.10

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

It has been possible to run Renovate against our setup where we combine local helm dependency charts (without `repository` field) and an external dependency chart (with repository). The latter has been managed by Renovate, the former has been ignored. Both are [allowed for helmv3](https://helm.sh/docs/helm/helm_dependency_update/)

With the help of the debug logs have identified this changes of PR https://github.com/renovatebot/renovate/pull/13937 to be the origin of our trouble. Since than the code cannot handle that `repository` can be undefined.

If I got the local development environment running on my Windows machine, I would try to add a test to the suite which represents a local helm dependency chart as prerequisite to compare the situation before and after the PR. I think it is quite simple to fix. The filter condition https://github.com/renovatebot/renovate/blob/fd423903f012f62f912dd9c795650c1ff44e67b7/lib/manager/helmv3/utils.ts#L59 is new to the stream and the implementation of `isAlias` does not handle undefined repositories:
https://github.com/renovatebot/renovate/blob/fd423903f012f62f912dd9c795650c1ff44e67b7/lib/manager/helmv3/utils.ts#L76

### Relevant debug logs

<details><summary>Logs</summary>

```
TypeError: Cannot read property 'startsWith' of undefined
    at isAlias (/usr/src/app/node_modules/renovate/lib/manager/helmv3/utils.ts:76:21)
    at /usr/src/app/node_modules/renovate/lib/manager/helmv3/utils.ts:59:30
    at Array.filter (<anonymous>)
    at getRepositories (/usr/src/app/node_modules/renovate/lib/manager/helmv3/utils.ts:59:6)
    at updateArtifacts (/usr/src/app/node_modules/renovate/lib/manager/helmv3/artifacts.ts:140:41)
    at getUpdatedPackageFiles (/usr/src/app/node_modules/renovate/lib/workers/branch/get-updated.ts:234:23)
    at processBranch (/usr/src/app/node_modules/renovate/lib/workers/branch/index.ts:361:17)
    at writeUpdates (/usr/src/app/node_modules/renovate/lib/workers/repository/process/write.ts:38:17)
    at update (/usr/src/app/node_modules/renovate/lib/workers/repository/process/extract-update.ts:129:11)
    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:46:17)
    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:148:7)
    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

### PR DESCRIPTION
**Pull request by secustor**
## Changes
- ignore dependencies which have not defined `repository` and therefore use the implied file structure 
- handle null and undefined as parameter for `isAlias`

## Context
Closes #14261

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository
https://github.com/secustor/14261

### CODE REVIEW

File: lib/manager/helmv3/utils.spec.ts [hunk 1]
```diff
@@ -39,5 +39,19 @@ describe('manager/helmv3/utils', () => {
       });
       expect(repository).toBe(repoUrl);
     });
+
+    it('return false if repository is null', () => {
+      const repository = resolveAlias(null, {
+        anotherRepository: 'https://charts.helm.sh/stable',
+      });
+      expect(repository).not.toBeTrue();
```
Comment (viceice) [inline lib/manager/helmv3/utils.spec.ts:47]:
```suggestion
      expect(repository).toBeFalse();
```

File: lib/manager/helmv3/utils.spec.ts [hunk 2]
```diff
@@ -39,5 +39,19 @@ describe('manager/helmv3/utils', () => {
...
+    });
+
+    it('return false if repository is undefined', () => {
+      const repository = resolveAlias(undefined, {
+        anotherRepository: 'https://charts.helm.sh/stable',
+      });
+      expect(repository).not.toBeTrue();
```
Comment (viceice) [inline lib/manager/helmv3/utils.spec.ts:54]:
```suggestion
      expect(repository).toBeFalse();
```

**Review by viceice — COMMENTED** [review]
Otherwise LGTM

**Comment by viceice** [comment]
We've now failing tests, seems there us a wrong return value somewhere.

**Comment by secustor** [comment]
Yes, I have misread `resolveAlias` as `isAlias`. It is  fixed now

File: lib/manager/helmv3/utils.ts [hunk 3]
```diff
@@ -56,6 +56,7 @@ export function resolveAlias(
 export function getRepositories(definitions: ChartDefinition[]): Repository[] {
```
Comment (viceice) [inline lib/manager/helmv3/utils.ts:56]: Line 45 should probably return `null`

**Review by viceice — APPROVED** [review]

**Pull request merged**